### PR TITLE
Feature/fga/auto accept invite

### PR DESCRIPTION
**Pull request by ganfra**
This PR introduces AutoAcceptInvite. It can be enabled at compile time for now.

### CODE REVIEW

File: vector/src/main/java/im/vector/app/AppStateHandler.kt [hunk 1]
```diff
@@ -117,16 +123,11 @@ class AppStateHandler @Inject constructor(
         return (selectedSpaceDataSource.currentValue?.orNull() as? RoomGroupingMethod.ByLegacyGroup)?.groupSummary?.groupId
     }
 
-    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
-    fun entersForeground() {
-    }
-
     @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
     fun entersBackground() {
-        compositeDisposable.clear()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/AppStateHandler.kt:126]: Never disposed then? this change is a bit weird here.
Maybe move the code from `init {}` to `entersForeground()`?

File: vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt [hunk 2]
```diff
@@ -57,10 +57,6 @@ class CallUserMapper(private val session: Session, private val protocolsChecker:
                 // will make sure we know where how to map calls and also allow us know not to display
                 // it in the future.
                 invitedRoom.markVirtual(nativeRoomId)
-                // also auto-join the virtual room if we have a matching native room
-                // (possibly we should only join if we've also joined the native room, then we'd also have
-                // to make sure we joined virtual rooms on joining a native one)
-                session.joinRoom(invitedRoomId)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt:63]: test the compil flag?

File: vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt [hunk 3]
```diff
@@ -204,21 +206,25 @@ class HomeDetailViewModel @AssistedInject constructor(@Assisted initialState: Ho
...
-                            val dmInvites = session.getRoomSummaries(
-                                    roomSummaryQueryParams {
-                                        memberships = listOf(Membership.INVITE)
-                                        roomCategoryFilter = RoomCategoryFilter.ONLY_DM
-                                        activeSpaceFilter = activeSpaceRoomId?.let { ActiveSpaceFilter.ActiveSpace(it) } ?: ActiveSpaceFilter.None
-                                    }
-                            ).size
+                            var dmInvites = 0
+                            var roomsInvite = 0
+                            if (!autoAcceptInvites.hideInvites) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt:211]: Could be nicer if we also had a method `autoAcceptInvites.showInvites()`

File: vector/src/main/java/im/vector/app/features/invite/AutoAcceptInvites.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+
+package im.vector.app.features.invite
+
+import javax.inject.Inject
+
+interface AutoAcceptInvites {
+    val isEnabled: Boolean
+    val hideInvites: Boolean
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/invite/AutoAcceptInvites.kt:23]: Default could be `= isEnabled` in the itf, no?
Also would be nice to add some Kdoc

**Review by bmarty — APPROVED** [review]
LGTM, just some minor remarks

File: vector/src/main/java/im/vector/app/AppStateHandler.kt (hunk 1 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/AppStateHandler.kt:126]: Nope I want the invites to be accepted when the app is in background so I can receive calls.
Its just to keep a reference to the disposable so it doesn't get GC

File: vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt (hunk 2 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt:63]: I don't think so, auto join virtual room is not mandatory

File: vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt (hunk 3 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt:211]: Yes

File: vector/src/main/java/im/vector/app/features/invite/AutoAcceptInvites.kt (hunk 4 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/invite/AutoAcceptInvites.kt:23]: Yes will add some.

File: vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/call/lookup/CallUserMapper.kt:63]: If you say so :)

File: vector/src/main/java/im/vector/app/AppStateHandler.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/AppStateHandler.kt:126]: So you will restore this line?

Comment (bmarty) [reply vector/src/main/java/im/vector/app/AppStateHandler.kt:126]: Thanks!

**Review by bmarty — APPROVED** [review]

**Pull request merged**